src/search/cost_test_ebpf.cc

```diff
@@ -214,8 +299,211 @@ void test1() {
   mem_t::_layout.clear();
 }
 
+//Only used while initially trying to modify error_cost function and check it
+//Not applicable anymore because the error cost is back to original
+//For repair new error cost calculation functions added which are tested from test 4 onwards
+void test3(){
+
+  //inst p1[N3], p2[N3];
+  //N3: constant
+  //generate new program length
+  //both should be same length
+  //p1: orig, p2: synth
+  //synth not easy, doesnt pass static check
+  //provide a safe program for p1
+  //can use win_select_test_ebpf line 60 
+
+
+  inst p1[] = {inst(NOP), 
+                inst(MOV64XC, 0, 0),
+                 inst(EXIT),
+                };
+
+  inst p2[] = {inst(AND64XC, 1, 0), 
+                inst(MOV64XC, 0, 0),
+                 inst(EXIT),
+                };
+  //part to be verified
+  //only this interval's code is different between p1 and p2
+  //Other parts (code outside the window) of p1 and p2 are identical
+  //make sure unsafe instruction is in the window
+  //wind_start can be from 0 instruction index
+  int win_start = 0, win_end = 0;
+
+  mem_t::_layout.clear();
+  
+  inst::max_prog_len = sizeof(p1) / sizeof(inst);
+
+  //inst_var.h (line 46): PGM_INPUT types for different kinds of program inputs
+  mem_t::set_pgm_input_type(PGM_INPUT_pkt);
+  //we don't use packet in the above programs
+  mem_t::set_pkt_sz(4);
+  //mem_t::add_map(map_attr(128, 64, 91));
+  //mem_t::add_map(map_attr(96, 96, 91));
+  //mem_t::add_map(map_attr(64, 128, 91));
+  mem_t::_layout._n_randoms_u32 = 1;
+  smt_var::init_static_variables();
+
+  print_test_res(get_safety_cost(p1, p2, win_start, win_end) == ERROR_COST_MAX, "cost:error change 1 check");
```

Comment:
Is this supposed to test safety cost repair?

Reply:
Hey Qiongwen, so this is 'get_safety_cost' only. Initially while we were changing the error_cost function directly, I created the 'get_safety_cost' function and used it in this test case. However, now the error_cost is back to original and we added error_cost_repair and safety_cost repair. So, I don't think this test you are referring to and 'get_safety_cost' is needed anymore. I took them out of the code